adoc/chapters/programming_interface.adoc

```diff
@@ -9453,8 +9453,10 @@ a@
 ----
 size_type max_size() const noexcept
 ----
-   a@ Returns the maximum number of elements any accessor of this type would be
-      able to access.
+   a@ Deprecated.
```

Comment:
```suggestion
   a@ Deprecated by SYCL 2020.
```

I think we need to be clear about which version of the spec deprecates this API.  Otherwise, it won't be clear whether it is deprecated in SYCL 2020 vs. SYCL Next (once we publish SYCL Next).